packages/eslint-plugin/tests/rules/naming-convention/naming-convention.test.ts

```diff
@@ -13,6 +16,38 @@ const parserOptions = {
   project: './tsconfig.json',
 };
 
+/** Union of all case error configs with `messageId` as discriminant property for type narrowing */
+type CaseErrorWithDataConfig =
+  | {
+      messageId: 'doesNotMatchFormat';
+      type: Capitalize<SelectorsString>;
+      name: string;
+      expectedFormats: PredefinedFormatsString[];
+    }
+  | { messageId: 'unexpectedUnderscore' }; // todo add more detail and union members as required
+
+function createExpectedTestCaseErrorsWithData(
+  caseErrorConfigs: CaseErrorWithDataConfig[],
+): TestCaseError<MessageIds>[] {
```

Comment:
We generally try to avoid test case generators like this. They obscure the test cases (which is already an issue in the `naming-convention` tests tbh), and also make it harder for our autoformatting on test cases to detect strings that should be formatted. 

Since this function outputs an object with roughly the same number of properties, I'd say let's remove it & inline the test cases directly.

Reply:
I don't have a good enough understanding of the repo so it took some time to figure out the available valid options for these assertions so thought typing it would help the next person.

As you say, the function outputs roughly the same object, it was mostly about the developer experience, but will remove